### DISCUSSION TITLE
How do I allow large uploads?

### DISCUSSION THREAD

**Discussion by DwifteJB** [nitrojs/nitro#2673] [Q&A]
I want to allow users to upload multiple mp3 files, up to around 3gb, but I have no clue how id allow multi-part uploads, would a good way to do this be through web sockets?

**Comment by DwifteJB** [comment]
https://gist.github.com/ChrisGV04/8350b53958f207ef2d33a9ea96ecd882

I've found this from @ChrisGV04, but I need to pass additional information, such as the playlist name, playlist album art, etc etc

**Reply by ChrisGV04** [reply]
Hi @DwifteJB !
You can certainly do that by adding extra fields to the `FormData` object and use them at your API endpoint.

The gist you are referring to is only considering the "files" and "file" keys from the `FormData`, but you can also add the playlist name and other info to it.

```ts
// Add extra fields to FormData like this
fd.set("name", "My new playlist");
```

```ts
// Get FormData fields on the server
const data = await readFormData(event);

// Read specific field
const playlistName = data.get("name"); // Make sure to validate the input server-side as well
```

**Reply by ChrisGV04** [reply]
And just in case... the gist only supports streaming uploads. If that doesn't work for you and you need multi-part file uploads, I linked the NuxtHub implementation on my original reply:

https://github.com/unjs/nitro/discussions/721#discussioncomment-10208008

In that case, you might need to have one API endpoint to create the database entry for the playlist with name, art, etc. and another endpoint to upload the parts of the mp3 files.

**Comment by stafyniaksacha** [comment]
I did not see anything related to nitro yet, but TUS protocol is really great for large file uploads: https://github.com/tus/tus-node-server
